libdevice/cmake/modules/SYCLLibdevice.cmake

```
@@ -350,55 +348,69 @@ check_cxx_compiler_flag(-Wno-invalid-noreturn HAS_NO_INVALID_NORETURN_WARN_FLAG)
 # ones.
 add_devicelibs(libsycl-itt-stubs
   SRC itt_stubs.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${itt_obj_deps})
 add_devicelibs(libsycl-itt-compiler-wrappers
   SRC itt_compiler_wrappers.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${itt_obj_deps})
 add_devicelibs(libsycl-itt-user-wrappers
   SRC itt_user_wrappers.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${itt_obj_deps})
 
 add_devicelibs(libsycl-crt
   SRC crt_wrapper.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${crt_obj_deps}
   EXTRA_OPTS $<$<BOOL:${HAS_NO_INVALID_NORETURN_WARN_FLAG}>:-Wno-invalid-noreturn>)
 
 add_devicelibs(libsycl-complex
   SRC complex_wrapper.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${complex_obj_deps})
 add_devicelibs(libsycl-complex-fp64
   SRC complex_wrapper_fp64.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${complex_obj_deps} )
 add_devicelibs(libsycl-cmath
   SRC cmath_wrapper.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${cmath_obj_deps})
 add_devicelibs(libsycl-cmath-fp64
   SRC cmath_wrapper_fp64.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${cmath_obj_deps} )
+set(imf_build_archs)
 add_devicelibs(libsycl-imf
   SRC imf_wrapper.cpp
-  DEPENDENCIES ${imf_obj_deps})
+  DEPENDENCIES ${imf_obj_deps}
+  BUILD_ARCHS ${imf_build_archs})
 add_devicelibs(libsycl-imf-fp64
   SRC imf_wrapper_fp64.cpp
-  DEPENDENCIES ${imf_obj_deps})
+  DEPENDENCIES ${imf_obj_deps}
+  BUILD_ARCHS ${imf_build_archs})
 add_devicelibs(libsycl-imf-bf16
   SRC imf_wrapper_bf16.cpp
-  DEPENDENCIES ${imf_obj_deps})
+  DEPENDENCIES ${imf_obj_deps}
+  BUILD_ARCHS ${imf_build_archs})
 add_devicelibs(libsycl-bfloat16
   SRC bfloat16_wrapper.cpp
+  BUILD_ARCHS ${full_build_archs}
   DEPENDENCIES ${cmath_obj_deps})
 if(MSVC)
   add_devicelibs(libsycl-msvc-math
     SRC msvc_math.cpp
+    BUILD_ARCHS ${full_build_archs}
     DEPENDENCIES ${cmath_obj_deps})
 else()
   if(UR_SANITIZER_INCLUDE_DIR)
+    set(sanitizer_build_archs "")
```

NIT: to be consistent with `full_build_archs`, and in general IMO this conveys better that the variable is an empty list and not an empty string.
```suggestion
    set(sanitizer_build_archs)
```